android/modules/calendar/src/java/ti/modules/titanium/calendar/CalendarModule.java

```diff
@@ -143,9 +144,11 @@ public KrollPromise<KrollDict> requestCalendarPermissions(
 			TiBaseActivity.registerPermissionRequestCallback(TiC.PERMISSION_CODE_CALENDAR, permissionCallback,
 				callbackThisObject, promise);
 			Activity currentActivity = TiApplication.getInstance().getCurrentActivity();
-			currentActivity.requestPermissions(
-				new String[] { Manifest.permission.READ_CALENDAR, Manifest.permission.WRITE_CALENDAR },
-				TiC.PERMISSION_CODE_CALENDAR);
+			if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
```

Comment:
The `hasCalendarPermissions()` method just few lines above it already handles sdk-version < 23, so it'll always be `true` here.